Reformat proxy docs refresh (#18623)
hashicorp/consul: website/content/docs/connect/config-entries/proxy-defaults.mdx

#### Values

- Default: `default`
- Data type: String

### `spec`

Map that contains the details about the ProxyDefaults configuration entry. The `apiVersion`, `kind``, and `metadata` fields are siblings of the `spec` field. All other configurations are children.

### `spec.config`

Specifies an arbitrary map of configuration values used by service mesh proxies. The available configurations depend on the mesh proxy you use. You can configure any global values that your proxy allows in this field. Refer to the following topics for additional information:

- [Envoy proxy configuration option](/consul/docs/connect/proxies/envoy#proxy-config-options)
- [Built-in proxy configuration options](/consul/docs/connect/proxies/built-in#proxy-config-key-reference) 

#### Values

- Default: None
- Data type: Map 

### `spec.envoyExtensions`

Specifies a list of extensions that modify Envoy proxy configurations. Refer to [Envoy extensions](/consul/docs/connect/proxies/envoy-extensions) for additional information.

#### Values

- Default: None

> - Data type: Map of containing the following fields:

Author: ```suggestion
- Data type: List of maps of containing the following fields:
```

| `envoyVersion` | Specifies the version of Envoy that the extension is allowed to work with. Consul validates the version during xDS updates. If a different version is in use, Consul skips the extension and writes the event to the log. <p>The `consulVersion` and `envoyVersion` must both validate for Consul to implement the extension.</p> | String | None |

### `spec.mode`

Specifies a mode for how proxies direct inbound and outbound traffic. You can specify one of the following values:

- `transparent`: In transparent mode, proxies capture and redirect inbound and outbound traffic. The mode does not enable traffic redirection, but directs Consul to configure Envoy as if traffic is already being redirected. 

- `direct`: In this mode, the local application and other proxies must directly dial proxy listeners. 

#### Values

- Default: None
- Data type: String 

### `spec.transparentProxy`

> Contains configurations for proxies that are running in transparent proxy mode. Refer to [Transparent proxy mode](/consul/docs/k8s/connect/transparent-proxy) for additional information. 

Author: ```suggestion
Contains configurations for proxies that are running in transparent proxy mode. This mode enables permissive mTLS for Consul so that you can use your Kubernetes cluster's DNS service instead of Consul DNS. Refer to [Transparent proxy mode](/consul/docs/k8s/connect/transparent-proxy) for additional information. 
```

#### Values

- Default: None
- Data type: Map of containing the following fields:
   - `outboundListenerPort`
   - `dialedDirectly`

The following table describes how to configure values in the `TransparentProxy` map:

| Parameter | Description | Data type | Default |
| ---       | ---         | ---       | ---     |

> | `outboundListenerPort` | Specifies the port that the proxy listens on for outbound traffic. Outbound application traffic must be  captured and redirected to this port. | Integer | `15001` |

Author: ```suggestion
| `outboundListenerPort` | Specifies the port that the proxy listens on for outbound traffic. Outbound application traffic must be captured and redirected to this port. | Integer | `15001` |
```

> | `dialedDirectly` | Determines whether other proxies in transparent mode can directly dial this proxy instance's IP address. Proxies in transparent mode commonly dial upstreams at the [`virtual`](/consul/docs/services/configuration/services-configuration-reference#tagged_addresses-virtual) tagged address, which load balances across instances. Dialing individual instances can be helpful when sending requests to stateful services, such as database clusters with a leader. | Boolean | `false` |

Author: ```suggestion
| `dialedDirectly` | Determines whether other proxies in transparent mode can directly dial this proxy instance's IP address. Proxies in transparent mode commonly dial upstreams at the [`virtual` tagged address](/consul/docs/services/configuration/services-configuration-reference#tagged_addresses-virtual), which load balances across instances. Dialing individual instances can be helpful when sending requests to stateful services, such as database clusters with a leader. | Boolean | `false` |
```

- `local`: The proxy makes an outbound connection to a gateway running in the same datacenter.
- `remote`: The proxy makes an outbound connection to a gateway running in the destination datacenter.

#### Values

- Default: None
- Data type: Map containing the `mode` field

### `spec.expose`

Specifies default configurations for exposing HTTP paths through Envoy. Exposing paths through Envoy enables services to protect themselves by only listening on `localhost`. Applications that are not Consul service mesh-enabled are still able to contact an HTTP endpoint. 

Example use-cases include exposing the `/metrics` endpoint to a monitoring system, such as Prometheus, and exposing the `/healthz` endpoint to the kubelet for liveness checks. Refer to [Expose Paths Configuration Reference](/consul/docs/connect/proxy-config-reference#expose-paths-configuration-reference) for additional information.

#### Values

- Default: None
- Data type: Map containing the following parameters:
   - [`checks`](#expose-checks)
   - [`aths`](#expose-paths)

### `spec.expose{}.checks`

> Exposes all HTTP and gRPC checks registered with the agent when set to `true`. Envoy exposes listeners for the checks and only accepts connections originating from localhost or Consul's [`advertise_addr`](/consul/docs/agent/config/config-files#advertise). The ports for the listeners are dynamically allocated from the agent's [`expose_min_port`](/consul/docs/agent/config/config-files#expose_min_port) and [`expose_max_port`](/consul/docs/agent/config/config-files#expose_max_port) configurations.

Author: ```suggestion
Exposes all HTTP and gRPC checks registered with the agent when set to `true`. Envoy exposes listeners for the checks and only accepts connections originating from localhost or the [Consul agent's `advertise_addr`](/consul/docs/agent/config/config-files#advertise). The ports for the listeners are dynamically allocated from the [agent's `expose_min_port`](/consul/docs/agent/config/config-files#expose_min_port) and [`expose_max_port`](/consul/docs/agent/config/config-files#expose_max_port) configurations.
```

> We recommend enabling the `Checks` configuration when a Consul client cannot reach registered services over localhost, such as when Consul agents run in their own pods in Kubernetes.

Author: ```suggestion
We recommend enabling the `Checks` configuration when a Consul client cannot reach registered services over localhost.
```

#### Values

- Default: `false`

> - Data type: boolean

Author: ```suggestion
- Data type: Boolean
```
